src/finance_quote/modules/fixerio.py

```diff
@@ -0,0 +1,185 @@
+
+        self.logger.debug("Latest prices downloaded.")
+
+        # Since these are daily rates, cache them into a file.
+        # Ignored for now since we are downloading individual currencies, not all together.
+        self.__save_rates(result)
+
+        return result
+
+    def get_yesterdays_file_path(self):
+        """ Full path to the today's rates file. """
+        from pydatum import Datum
+        datum = Datum()
+        datum.yesterday()
+        yesterday = datum.get_iso_date_string()
+
+        return self.__get_rate_file_path(yesterday)
+
+    def get_todays_file_path(self):
+        """ path to today's cached file """
+        from pydatum import Datum
+        datum = Datum()
+        datum.today()
+        today = datum.get_iso_date_string()
+
+        return self.__get_rate_file_path(today)
+
+    def latest_rates_exist(self):
+        """ Check if latest rates cached file exists. """
+        file_path = self.get_todays_file_path()
+        exists = os.path.isfile(file_path)
+
+        if exists:
+            self.logger.debug("Cached file found at " + file_path)
+
+        return exists
+
+    def __get_rate_file_path(self, filename):
+        """
+        Assemble full file path for the given name (date).
+        """
+        return os.path.abspath("{cache_path}/{filename}.json".format(
+            cache_path=self.cache_path, filename=filename))
+
+    def __read_rates_from_file(self):
+        file_path = self.get_todays_file_path()
+
+        with open(file_path, 'r') as file:
+            content = file.read()
+
+        # self.logger.debug(f"cached rates: {content}")
+        return json.loads(content)
+
+    def __save_rates(self, rates):
+        """
+        Saves the retrieved rates into a cache file
+        """
+        file_date = rates["date"]
+        filename = self.__get_rate_file_path(file_date)
+
+        content = json.dumps(rates)
+
+        with open(filename, 'w') as file:
+            file.write(content)
+            self.logger.debug("rates saved in " + filename + ".")
+
+
+class FixerioModelMapper:
+    """ Maps the result from Fixer.io into an array of FixerioQuote """
+    def __init__(self, json_response):
+        self.__data = json_response
+
+    def get_model_for_symbol(self, symbol: str) -> FixerioQuote:
+        """ Read and map a single currency rate """
+        from datetime import datetime
+
+        date_str = self.__data["date"]
+        rate_date = datetime.strptime(date_str, "%Y-%m-%d")
+        assert isinstance(rate_date, datetime)
+
+        base = self.__data["base"]
+        rates_dict = self.__data["rates"]
+        value_str = rates_dict[symbol]
+        value = Decimal(value_str)
+        # The rate is inverse value.
+        rate = Decimal(1) / value
+        # Round to 6 decimals max.
+        rounded_str = "{0:.6f}".format(rate)
+        rounded = Decimal(rounded_str)
+
+        model = Quote()
+        model.namespace = "CURRENCY"
```

Comment:
better to give the parameters in the constructor than changing them afterwards (not very pythonic)